OneLogin IAM CCP Connector Migration

v-gudivya opened the pull request:
Required items, please complete
   
   Change(s):
   - Added new CCP based connector for OneLogin IAM

   Reason for Change(s):
   - Due to deprecation of the existing Azure Function App based connector.

   Version Updated:
   - Yes

   Testing Completed:
   - Yes
 
[Checklist](https://microsoftapc-my.sharepoint.com/:x:/g/personal/v-gudivya_microsoft_com/EUmMMQdlN4BDjsi4vhQjKewBtgafYjzjx8XuIya6v5tGsQ?e=BVHr78)

![OneLogin Connection](https://github.com/user-attachments/assets/1754d650-68b0-4635-85f7-729a22d4a2a4)
![OneLogin Graph](https://github.com/user-attachments/assets/3f352bba-3ad6-4a75-b2cf-a8bb81718884)
![OneLogin UI](https://github.com/user-attachments/assets/f09f15bb-323a-427e-98a2-c846f4c816fc)
![OneLogin Event Logs](https://github.com/user-attachments/assets/f6a51d00-e5ec-46ce-9ed9-64dfc35bbfee)
![OneLogin User Logs](https://github.com/user-attachments/assets/64394f27-431f-4d40-bca4-32e6fe002e12)

v-gudivya committed:
Create ConnectorDefinition file

v-gudivya committed:
updating the files

v-gudivya committed:
updated files

v-gudivya committed:
updated solution file

v-gudivya committed:
updated files

v-gudivya committed:
file updated

v-gudivya committed:
updating latest files

v-gudivya committed:
latest file templates

v-gudivya committed:
new file

v-gudivya committed:
updated the files

v-pmalreddy committed:
Updated KQL Validation tests

manishkumar1991 commented:
@v-gudivya  Why Armorblox changes are present in this PR
![image](https://github.com/user-attachments/assets/8a4969ab-7e89-4980-9d3f-f677863288fe)

manishkumar1991 commented on Solutions/OneLoginIAM/Data Connectors/OneLoginIAMLogs_ccp/OneLoginIAMLogs_DCR.json:
```diff
@@ -0,0 +1,167 @@
+                "Custom-OneLoginEventsV2_CL": {
+                    "columns": [
+                        {
+                            "name": "TimeGenerated",
```
are you getting this field "TimeGenerated" in response ? if not kindly remove it

v-gudivya replied:
Hi Manish, removed the TimeGenerated field from the 'Columns' section of the DCR and mapped it in the KQL query. I will be committing the new changes.

manishkumar1991 commented on Solutions/OneLoginIAM/Data Connectors/OneLoginIAMLogs_ccp/OneLoginIAMLogs_DCR.json:
```diff
@@ -0,0 +1,167 @@
+                            "type": "string"
+                        },
+                        {
+                            "name": "['title']",
```
Why the name of field is mentioned like that ['title']

Can you share the api response for this ?

v-gudivya replied:
Hi Manish, title is a reserved word, and it was throwing an error while connecting. When verified at scuba end, identified that we have to define as ['title']. Anyhow I have changed it under the columns section of DCR and updated KQL accordingly.

![title word error](https://github.com/user-attachments/assets/e861af54-e71b-40b0-ba0c-6176ed7095cd)

manishkumar1991 commented on Solutions/OneLoginIAM/Data Connectors/OneLoginIAMLogs_ccp/OneLoginIAMLogs_DCR.json:
```diff
@@ -0,0 +1,167 @@
+                    "destinations": [
+                        "clv2ws1"
+                    ],
+                    "transformKql": "source | project-rename EventId = id, AppName = app_name, AppId = app_id, RoleName = role_name, RoleId = role_id, PolicyName = policy_name, PolicyId = policy_id, PolicyType = policy_type, UserId = user_id, UserName = user_name, ActorSystem = actor_system, CustomMessage = custom_message, AccountId = account_id, Ipaddr = ipaddr, ActorUserName = actor_user_name,  ...
```
TimeGenerated Field is essential/important here kindly map that in transform kql

v-gudivya replied:
Hi Manish, yes, TimeGenerated field is mapped in KQL query and removed from columns section of the DCR. I will be committing the new changes.

manishkumar1991 commented on Solutions/OneLoginIAM/Data Connectors/OneLoginIAMLogs_ccp/OneLoginIAMLogs_DCR.json:
```diff
@@ -0,0 +1,167 @@
+                    "destinations": [
+                        "clv2ws1"
+                    ],
+                    "transformKql": "source | project-rename UserAttributesUsername = username, UserAttributesFirstname = firstname, UserAttributesEmail = email, UserAttributesLastname = lastname, UserAttributesDepartment = department, UserAttributesOpenidName = openid_name, UserAttributesTitle = ['title'], UserAttributesId = id",
```
TimeGenerated Field is essential/important here kindly map that in transform kql

v-gudivya replied:
Hi Manish, yes, TimeGenerated field is mapped in KQL query and removed from columns section of the DCR. I will be committing the new changes.

manishkumar1991 commented on Solutions/OneLoginIAM/Data Connectors/OneLoginIAMLogs_ccp/OneLoginIAMLogs_DCR.json:
```diff
@@ -0,0 +1,167 @@
+                "Custom-OneLoginUsersV2_CL": {
+                    "columns": [
+                        {
+                            "name": "TimeGenerated",
```
are you getting this field "TimeGenerated" in response ? if not kindly remove it

v-rajmalviya committed:
Merge branch 'master' into v-gudivya-OneLoginIAM

v-gudivya committed:
Corrected the files

v-rajmalviya committed:
Update Utils.cs

Replaced Assembly.CodeBase with Assembly.Location to resolve a SYSLIB0012 warning introduced in newer .NET SDKs.

v-rajmalviya committed:
Update OneLoginIAMLogs_ConnectorDefinition.json

Changed domain value to fix validation

v-rajmalviya committed:
Removed auth URL from definition file to polling file

Removed auth URL from definition file to polling file

v-rajmalviya committed:
git commit -m "Merge completed for OneLogin IAM detection rules"
Merge branch 'master' into v-gudivya-OneLoginIAM

v-rajmalviya committed:
Merge branch 'master' into v-gudivya-OneLoginIAM

manishkumar1991 commented on Solutions/OneLoginIAM/Data Connectors/OneLoginIAMLogs_ccp/OneLoginIAMLogs_ConnectorDefinition.json:
```diff
@@ -0,0 +1,103 @@
+              "type": "Textbox",
+              "parameters": {
+                "label": "OneLogin Sub Domain",
+                "placeholder": "Enter your Company's OneLogin Sub Domain",
```
@v-gudivya

v-gudivya replied:
Hi Manish, Checking on it.

v-gudivya replied:
Hi @manishkumar1991 

Added a NOTE in the instruction steps of connector definition file for user understanding, also updated latest screenshot above.

v-gudivya committed:
Added a NOTE in Connector definition file

manishkumar1991 reviewed (dismissed):
Approved

v-gudivya committed:
Multiple Domain Connection Changes

v-gudivya committed:
Updated the files

v-gudivya committed:
Merge branch 'master' into v-gudivya-OneLoginIAM

manishkumar1991 commented on Solutions/OneLoginIAM/Data Connectors/OneLoginIAMLogs_ccp/OneLoginIAMLogs_DCR.json:
```diff
@@ -0,0 +1,159 @@
+                    "destinations": [
+                        "clv2ws1"
+                    ],
+                    "transformKql": "source | extend TimeGenerated = now() | project-rename EventId = id, AppName = app_name, AppId = app_id, RoleName = role_name, RoleId = role_id, PolicyName = policy_name, PolicyId = policy_id, PolicyType = policy_type, UserId = user_id, UserName = user_name, ActorSystem = actor_system, CustomMessage = custom_message, AccountId = account_id, Ipaddr = ipaddr, Ac ...
```
TimeGenerated  should be mapped with proper field , here in this case i can see that EventTimestamp  is perfect for mapping

v-gudivya replied:
Sure Manish, will work on it.

v-gudivya replied:
Hi @manishkumar1991, Mapped the TimeGenerated field in both the tables.

v-gudivya replied:
Hi @manishkumar1991, please find attachments below for the **TimeGenerated** mapping with **created_at** for both the tables. As per schema, created_at is in string data type, but I defined in datetime data type for human readable format.

![OneLogin_Events_TimeGenerated_Mapping](https://github.com/user-attachments/assets/fd69e72a-affb-43cc-9f14-fba09572a78a)
![OneLogin_API_Response_Events_TimeGenerated_Mapping](https://github.com/user-attachments/assets/9b991658-1134-4641-b3d5-9c554d409da6)
![OneLogin_Users_TimeGenerated_Mapping](https://github.com/user-attachments/assets/69aa5ed7-fbc9-49c2-9ec0-2fb22d5091df)
![OneLogin_API_Response_Users_TimeGenerated_Mapping](https://github.com/user-attachments/assets/ddec048e-8dfb-4b0e-9708-12848d8dd9c4)

v-gudivya committed:
TimeGenerated field is mapped

Modified the files with few changes and re-packaged.

v-gudivya committed:
Merge branch 'master' into v-gudivya-OneLoginIAM

manishkumar1991 reviewed (approved):
approved
